gateway/ln-gateway/src/rpc/mod.rs

```diff
@@ -74,6 +74,13 @@ pub struct RestorePayload {
     pub federation_id: FederationId,
 }
 
+#[derive(Debug, Serialize, Deserialize)]
+pub struct LightningReconnectPayload {
```

Comment:
Since payload is only `Some` when we are looking to reconnect to a node which potentially replaces the one that gatewayd currently knows, maybe we could rename this payload to `ReplaceLightningPayload`